lib/propolis/src/hw/virtio/viona.rs

```diff
@@ -56,9 +57,27 @@ impl PciVirtioViona {
         queue_size: u16,
         vm: &VmmHdl,
     ) -> io::Result<Arc<PciVirtioViona>> {
-        let dlhdl = dladm::Handle::new()?;
-        let info = dlhdl.query_vnic(vnic_name)?;
-        let hdl = VionaHdl::new(info.link_id, vm.fd())?;
+        let info =
+            libnet::get_link(&libnet::LinkHandle::Name(vnic_name.to_string()))
+                .map_err(|e| {
+                    io::Error::new(
+                        io::ErrorKind::Other,
+                        format!("failed to get link info: {e}"),
+                    )
+                })?;
+
+        let mtu = if let Some(mtu) = info.mtu {
+            Some(u16::try_from(mtu).map_err(|_| {
+                io::Error::new(
+                    io::ErrorKind::Other,
+                    format!("MTU too large: {mtu}"),
+                )
+            })?)
+        } else {
+            None
+        };
```

Comment:
If we want to keep propolis able to build (albeit useless binaries) on non-illumos platforms, perhaps we can wrap the libnet stuff pulled out into a function in `cfg` blocks?